lib/view_model/anon_invoice_page_view_model.dart

```diff
@@ -0,0 +1,192 @@
+  String receipientEmail;
+
+  @observable
+  String receipientName;
+
+  @observable
+  String description;
+
+  @observable
+  String amount;
+
+  @observable
+  ExecutionState state;
+
+  @computed
+  int get selectedCurrencyIndex => currencies.indexOf(selectedCurrency);
+
+  @observable
+  double? minimum;
+
+  @observable
+  double? maximum;
+
+  @action
+  void selectCurrency(Currency currency) {
+    selectedCurrency = currency;
+    maximum = minimum = null;
+    if (currency is CryptoCurrency) {
+      cryptoCurrency = currency;
+    }
+    _fetchLimits();
+  }
+
+  @action
+  Future<void> createInvoice() async {
+    state = IsExecutingState();
+    if (amount.isNotEmpty) {
+      final amountInCrypto = double.parse(amount);
+      if (minimum != null && amountInCrypto < minimum!) {
+        state = FailureState('Amount is too small');
+        return;
+      }
+      if (maximum != null && amountInCrypto > maximum!) {
+        state = FailureState('Amount is too big');
+        return;
+      }
+    }
+    final result = await anonPayApi.createInvoice(AnonPayRequest(
+      cryptoCurrency: cryptoCurrency,
+      address: address,
+      amount: amount.isEmpty ? null : amount,
+      description: description,
+      email: receipientEmail,
+      name: receipientName,
+      fiatEquivalent:
+          selectedCurrency is FiatCurrency ? (selectedCurrency as FiatCurrency).raw : null,
+    ));
+
+    _anonpayInvoiceInfoSource.add(result);
+
+    state = ExecutedSuccessfullyState(payload: result);
+  }
+
+  @action
+  void setRequestParams({
+    required String inputAmount,
+    required String inputName,
+    required String inputEmail,
+    required String inputDescription,
+  }) {
+    receipientName = inputName;
+    receipientEmail = inputEmail;
+    description = inputDescription;
+    amount = inputAmount;
+  }
+
+  @action
+  Future<void> generateDonationLink() async {
+    state = IsExecutingState();
+
+    final result = await anonPayApi.generateDonationLink(AnonPayRequest(
+      cryptoCurrency: cryptoCurrency,
+      address: address,
+      description: description,
+      email: receipientEmail,
+      name: receipientName,
+    ));
+
+    await sharedPreferences.setString(PreferencesKey.trocadorDonationLink, result.clearnetUrl);
+
+    state = ExecutedSuccessfullyState(payload: result);
+  }
+
+  Future<void> _fetchLimits() async {
+    final limit = await anonPayApi.fetchLimits(currency: cryptoCurrency);
+    minimum = limit.min;
+    maximum = limit.max != null ? limit.max! / 4 : null;
+    if (selectedCurrency is FiatCurrency) {
+      final fiatPrice = _fiatConversationStore.prices[cryptoCurrency];
+      maximum = maximum != null
+          ? double.tryParse(calculateFiatAmount(price: fiatPrice, cryptoAmount: maximum.toString()))
+          : null;
+      minimum = minimum != null
+          ? double.tryParse(calculateFiatAmount(price: fiatPrice, cryptoAmount: minimum.toString()))
+          : null;
+    }
```

Comment:
this check should be first, before making the API call, cause it won't be necessary to make the API call if the condition is true

Reply:
also, this will get the price for the fiat currency selected in the settings, not the one selected here